crates/common/src/da.rs

```diff
@@ -56,13 +55,12 @@ pub async fn sync_l1<Da>(
                     }
                 };
 
-            let mut queue = block_queue.lock().await;
-
-            if queue.len() < 10 {
-                queue.push_back(l1_block.clone());
-            } else {
-                debug!("Block queue is full, will try later...");
-                break;
```

Comment:
This check was replaced by having a channel which has a size of 10 instead of checking the queue length WHILE holding the lock.